src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDesignTimeDescriptor.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Razor.TagHelpers
+{
+    /// <summary>
+    /// A metadata class containing design time information about a tag helper.
+    /// </summary>
+    public class TagHelperDesignTimeDescriptor
+    {
+        /// <summary>
+        /// Instantiates a new instance of <see cref="TagHelperDesignTimeDescriptor"/>.
+        /// </summary>
+        /// <param name="summary">A summary on how to use a tag helper.</param>
+        /// <param name="remarks">Remarks on how to use a tag helper.</param>
+        /// <param name="outputElementHint">The HTML element a tag helper may output.</param>
+        public TagHelperDesignTimeDescriptor(string summary, string remarks, string outputElementHint)
```

Comment:
If others feel strongly about changing `TagHelperAttributeDesignTimeDescriptor` and `TagHelperDesignTimeDescriptor` to be mutable, suggest adding the `public` setters in this PR. Leave #441 to cover the existing `TagHelperAttributeDescriptor` and `TagHelperDescriptor`. This is possible / easy because the usage descriptors are only compared in test code.